x/metadata/keeper/msg_server.go

```diff
@@ -73,6 +122,31 @@ func (k msgServer) DeleteScope(
 	return types.NewMsgDeleteScopeResponse(), nil
 }
 
+func (k msgServer) ExpireScope(
```

Comment:
This `ExpireScope` msg and endpoint is exactly the same as the `DeleteScope` msg endpoint.

But there's an issue to address here. When a scope is deleted, any expiration entry for it should also processed. That should happen when invoked via the expiration module, or through the `DeleteScope` endpoint. In the end, the scope (and all it's stuff) should be deleted, it's deposit should be refunded, and the expiration record should also be deleted.

I _think_ this might be solved by updating the `DeleteScope` endpoint to first check for an expiration entry for the given id. If it exists, just invoke that entry. If there isn't an entry, then delete the scope as normal. Doing it this way would be a reason to have different `ExpireScope` and `DeleteScope` endpoints, because it would prevent an infinite loop. A drawback to this is that calling the `DeleteScope` endpoint might then emit the `ExpireScope` events.

Reply:
I added `RemoveExpiration` to the expiration keeper for modules to call and removed `ExpireScope` from metadata keeper. When `DeleteScope` is called it calls `RemoveExpiration` (returns `nil` if no expiration exists) and deletes the scope.